src/lib.rs

```diff
@@ -181,6 +181,26 @@ impl VersionInformation {
         self.eax
     }
 
+    /// Maximum number of addressable IDs for logical processors in this physical package*.
+    pub fn max_logical_processor_ids(self) -> Option<u8> {
+        if self.htt() {
+            Some((self.ebx >> 16) as u8)
+        } else {
+            None
+        }
+    }
+
+    /// This number is assigned to the local APIC on the processor during power up.
+    ///
+    /// This field was introduced in the Pentium 4 processor.
+    pub fn local_logical_processor_id(self) -> Option<u8> {
+        if self.htt() {
```

Comment:
Sorry, but I wasn't very clear in my original comment. In the manual, only *bits 23 - 16* have the footnote that they require `htt`, but the processor ID does not have the same restriction. Why does this need to be an `Option`?

Reply:
Yes, Intel manual said only the `max_logical_processor_ids` base on the `htt` flag, but when we use the `local_logical_processor_id`, it always need work together with `max_logical_processor_ids`. For example, I use a `Vec<T>`  to provides local storage for every core, it's meaningless if `max_logical_processor_ids` is `None` but `local_logical_processor_id ` is zero.

> The nearest power-of-2 integer that is not smaller than EBX[23:16] is the number of unique initial APIC IDs reserved for addressing different logical processors in a physical package. This field is only valid if CPUID.1.EDX.HTT[bit 28]= 1.

So, I choose to use `Option<u8>` for both fields, or we could give them default value (`max_logical_processor_ids` = 1, `local_logical_processor_id` = 0), what's your opinion?

> 28 HTT **Max APIC IDs reserved field is Valid.** A value of 0 for HTT indicates there is only a single logical processor in the package and software should assume only a single APIC ID is reserved. A value of 1 for HTT indicates the value in CPUID.1.EBX[23:16] (the Maximum number of addressable IDs for logical processors in this package) is valid for the package. 

> 22 ACPI **Thermal Monitor and Software Controlled Clock Facilities.** The processor implements internal MSRs that allow processor temperature to be monitored and processor performance to be modulated in predefined duty cycles under software control.

Reply:
Besides, the `local_logical_processor_id ` and `max_logical_processor_ids` only apply to non-NUMA desktop CPU, we need use Extended Topology Enumeration Leaf (0BH)  for more details, would you like me commit more changes about this?

[Intel® 64 Architecture Processor Topology Enumeration](https://software.intel.com/en-us/articles/intel-64-architecture-processor-topology-enumeration)

Reply:
> when we use the `local_logical_processor_id`, it always need work together with `max_logical_processor_ids`

Is that for your *specific* use or is that for *every* use? I agree that it seems like it might be silly to allow for one when you can't use the other, but such a decision should probably occur at a higher level than this library (such as when you are using it). If Intel introduced `local_logical_processor_id` before they introduced `max_logical_processor_ids`, then there must be *some* reason someone would use it by itself.

> I choose to use `Option<u8>` for both fields, or we could give them default value

Shouldn't `max` be an `Option` and `id` just a `u8`? No need for a default value, as far as I can tell.

Reply:
Ok, added some comment on it.